SilKit/source/services/logging/Logger.cpp

```diff
@@ -118,32 +313,124 @@ Logger::Logger(const std::string& participantName, Config::Logging config)
         {
 #if _WIN32
             auto stdoutSink = std::make_shared<spdlog::sinks::stdout_sink_mt>();
+
 #else
             auto stdoutSink = std::make_shared<spdlog::sinks::ansicolor_stdout_sink_mt>();
 #endif
-            stdoutSink->set_level(log_level);
-            _logger->sinks().emplace_back(std::move(stdoutSink));
+
+            if (sink.format == Config::Sink::Format::Json && sink.type != Config::Sink::Type::Remote)
+            {
+                using spdlog::details::make_unique; // for pre c++14
+                auto formatter = make_unique<spdlog::pattern_formatter>();
+
+                formatter->set_pattern(jsonpattern);
+                stdoutSink->set_formatter(std::move(formatter));
+                stdoutSink->set_level(log_level);
+                _loggerJson->sinks().emplace_back(std::move(stdoutSink));
+            }
+            else if(sink.type != Config::Sink::Type::Remote)
+            {
+                stdoutSink->set_level(log_level);
+                _loggerSimple->sinks().emplace_back(std::move(stdoutSink));
+            }
+
             break;
         }
         case Config::Sink::Type::File:
         {
-            //
             auto filename = fmt::format("{}_{:%FT%H-%M-%S}.txt", sink.logName, tmBuffer);
             auto fileSink = std::make_shared<spdlog::sinks::basic_file_sink_mt>(filename);
-            fileSink->set_level(log_level);
-            _logger->sinks().push_back(fileSink);
+
+            if (sink.format == Config::Sink::Format::Json)
+            {   
+                using spdlog::details::make_unique; // for pre c++14
+                auto formatter = make_unique<spdlog::pattern_formatter>();
+                formatter->set_pattern(jsonpattern);
+                fileSink->set_pattern(jsonpattern);
+                fileSink->set_level(log_level);
+                _loggerJson->sinks().push_back(fileSink);
+            }
+            else
+            {
+                fileSink->set_level(log_level);
+                _loggerSimple->sinks().push_back(fileSink);
+            }
         }
         }
     }
+    if (nullptr != _loggerSimple)
+    {
+        _loggerSimple->flush_on(to_spdlog(_config.flushLevel));
+    }
+    if (nullptr != _loggerJson)
+    {
+        _loggerJson->flush_on(to_spdlog(_config.flushLevel));
+    }
+}
+
+void Logger::Log(const LoggerMessage& msg)
+{
+    const auto now = log_clock::now();
+    if (nullptr != _loggerJson)
+    {
+        JsonLogMessage myJsonMsg(msg);
```

Comment:
Aggregate initialization ftw!

```suggestion
        JsonLogMessage myJsonMsg{msg};
```